components/net/http_loader.rs

```diff
@@ -437,6 +437,27 @@ fn no_referrer_when_downgrade_header(referrer_url: Url, url: Url) -> Option<Url>
     return strip_url(referrer_url, false);
 }
 
+/// https://w3c.github.io/webappsec-referrer-policy/#referrer-policy-strict-origin
+fn strict_origin(referrer_url: Url, url: Url) -> Option<Url> {
+    if referrer_url.scheme() == "https" && url.scheme() != "https" {
+        return None;
+    }
+    return strip_url(referrer_url, true);
+}
+
+/// https://w3c.github.io/webappsec-referrer-policy/#referrer-policy-strict-origin-when-cross-origin
+fn strict_origin_when_cross_origin(referrer_url: Url, url: Url) -> Option<Url> {
+    let cross_origin = referrer_url.origin() != url.origin();
+    if referrer_url.scheme() == "https" && url.scheme() != "https" {
+        return None;
+    } else {
+        if cross_origin {
+            return strip_url(referrer_url, true);
+        }
+        return strip_url(referrer_url, false);
```

Comment:
Nit: `return` is useless here and doing `strip_url(referrer_url, cross_origin)` is shorter.